src/connection.rs

```diff
@@ -155,6 +154,62 @@ impl Connection {
 			}
 		})
 	}
+
+	// TODO: do this without instantiating a DBus proxy evwry time.
+	/// Registers an events as defined in [`crate::events::names`]. This function registers a single event, like so:
+	/// ```rust
+	/// use atspi::events::names::ObjectEvents;
+	/// # tokio_test::block_on(async {
+	/// let connection = atspi::Connection::open().await.unwrap();
+	/// connection.register_event(ObjectEvents::StateChanged).await.unwrap();
+	/// # })
+	/// ```
+	///
+	/// # Errors
+	///
+	/// This function may return an error if it is unable to serialize the variant of the enum that has been passed (should never happen), or
+	/// a [`zbus::Error`] is caused by all the various calls to [`zbus::fdo::DBusProxy`] and [`zbus::MatchRule`].
+	pub async fn register_event<T: Signal + Serialize + std::fmt::Debug>(
+		&self,
+		event: T,
+	) -> Result<(), AtspiError> {
+		let dbus_proxy = DBusProxy::new(self.registry.connection()).await?;
+		let Ok(member_string) = serde_plain::to_string(&event) else {
+          return Err(AtspiError::ParseError("Unable to serialize member string to create a match rule."))
+        };
+		let match_rule = MatchRule::builder()
+			.msg_type(MessageType::Signal)
+			.interface(<T as Signal>::INTERFACE)?
+			.member(member_string)?
+			.build();
+		dbus_proxy.add_match_rule(match_rule).await?;
+		Ok(())
+	}
+
```

Comment:
I am not sure we want to hide the MatchRules like this.
This combines registering with registryd and adding a MatchRule to the bus, dbus-broker.
Two different daemons and separate connections.

Reply:
This is incorrect on two counts:

1. This *only* adds a match rule, it does not add anything to the registry. The registry doesn't seem to do anything. I have tested this pretty extensively, and I see nothing additional coming down the bus based on `registryd`; this seems to only by for key bindings at this point.
2. It uses the same connection; check `odilia/src/main.rs` lines 35-40.

This *does* hide match rules; but, of course, it does not stop a user from manually creating a DBusProxy, and adding a `MatchRule` manually. This is always an option. It seems to me though that is is unnecessary. The MatchRule can be safely hidden from the programmer.

It would also be fine if those types from `identify.rs` would have a static `Into<MatchRule>` implementation. Something like `identify::object::StateChangedEvent::into::<MatchRule>()`. This would allow the user of the library to know that the matchrule exists, and be able to modify it without needing to create it themselves.

Thoughts? I'd be fine with an implementation similar to that. I'll get working on it asap.

Reply:
>     1. This _only_ adds a match rule, it does not add anything to the registry. The registry doesn't seem to do anything.

[This is what I could find it does:](https://web.archive.org/web/20140414005650/http://www.linuxfoundation.org/collaborate/workgroups/accessibility/atk/at-spi/at-spi_on_d-bus#AT-SPI_D-Bus_Specification)

"
The registry daemon has two distinct tasks.

    Keeping track of all the accessible applications on the desktop.
    Dealing with and forwarding X events over AT-SPI D-Bus.
"

Reply:
> This is incorrect on two counts:

No more reviews after midnight for me.

> 
>     1. This _only_ adds a match rule, it does not add anything to the registry. 

Ah, yes. Right. I think the name made me think it did registering too.

> 
> This _does_ hide match rules; but, of course, it does not stop a user from manually creating a DBusProxy, and adding a `MatchRule` manually. This is always an option. It seems to me though that is is unnecessary. The MatchRule can be safely hidden from the programmer.

Again I think I jumped to the conclusion that it registered events because of its name.
It adds a MatchRule instead.

> 
> It would also be fine if those types from `identify.rs` would have a static `Into<MatchRule>` implementation. Something like `identify::object::StateChangedEvent::into::<MatchRule>()`. This would allow the user of the library to know that the matchrule exists, and be able to modify it without needing to create it themselves.

Sounds really good.

> 
> Thoughts? I'd be fine with an implementation similar to that. I'll get working on it asap.

You know we get `Into<MatchRule> for StateChangeEvent` for free if you implement `From<StateChangedEvent> for MatchRule`?

Reply:
Unfortunately, From seems to take a tegerence to self, which would require an instanciation of the type. In this case that is not necessary, but the MatchRuleBuilder functions aren't const so I'm just building an adsociated function to run at runtime.

The latest version of this PR implements a `match_rule()` and `match_rules()` associsted function for the struct snd enum containing it respectively. Check that out and let me know.